app/routes/surveys/components/AdminSideBar.js

```diff
@@ -6,61 +6,89 @@ import { Link } from 'react-router';
 
-const AdminSideBar = ({
-  surveyId,
-  deleteFunction,
-  actionGrant,
-  push
-}: Props) => {
-  const canEdit = actionGrant.includes('edit');
+type State = {
+  copied: boolean
+};
 
-  const onConfirm = () => {
-    if (deleteFunction && push) {
-      return (
-        deleteFunction &&
-        deleteFunction(surveyId).then(() => push && push('/surveys'))
-      );
-    }
-    if (deleteFunction) {
-      return deleteFunction && deleteFunction(surveyId);
-    }
-    return Promise.resolve();
+export class AdminSideBar extends React.Component<Props, State> {
+  state: State = {
+    copied: false
   };
 
-  return (
-    canEdit && (
-      <ContentSidebar className={styles.adminSideBar}>
-        <strong>Admin</strong>
-        <ul>
-          <li>
-            <Link to="/surveys/add">Ny undersøkelse</Link>
-          </li>
-          <li>
-            <Link to={`/surveys/${surveyId}/edit`}>Endre undersøkelsen</Link>
-          </li>
-          {deleteFunction && (
-            <ConfirmModalWithParent
-              title="Slett undersøkelse"
-              message="Er du sikker på at du vil slette denne undersøkelseen?"
-              onConfirm={onConfirm}
-            >
+  render() {
+    const { surveyId, deleteFunction, actionGrant, push, token } = this.props;
+
+    const canEdit = actionGrant.includes('edit');
+
+    const onConfirm = () => {
+      if (deleteFunction && push) {
+        return (
+          deleteFunction &&
+          deleteFunction(surveyId).then(() => push && push('/surveys'))
+        );
+      }
+      if (deleteFunction) {
+        return deleteFunction && deleteFunction(surveyId);
+      }
+      return Promise.resolve();
+    };
+
+    const shareLink = !token
+      ? ''
+      : `${config.webUrl}/surveys/${surveyId}/results/?token=${token}`;
+
+    return (
+      canEdit && (
+        <ContentSidebar className={styles.adminSideBar}>
+          <strong>Admin</strong>
+          <ul>
+            <li>
+              <Link to="/surveys/add">Ny undersøkelse</Link>
+            </li>
+            <li>
+              <Link to={`/surveys/${surveyId}/edit`}>Endre undersøkelsen</Link>
+            </li>
+            {deleteFunction && (
+              <ConfirmModalWithParent
+                title="Slett undersøkelse"
+                message="Er du sikker på at du vil slette denne undersøkelseen?"
+                onConfirm={onConfirm}
+              >
+                <li>
+                  <Link to="">Slett undersøkelsen</Link>
+                </li>
+              </ConfirmModalWithParent>
+            )}
+            {token && (
               <li>
-                <Link to="">Slett undersøkelsen</Link>
+                <CopyToClipboard
+                  text={shareLink}
+                  onCopy={() => this.setState({ copied: true })}
+                  style={{ marginTop: '5px' }}
+                >
+                  <Button>
+                    {this.state.copied ? 'Kopiert!' : 'Kopier delbar link'}
```

Comment:
We should support enable/disable this public sharing. We should also support invalidating the old token.

Reply:
😞